src/main/java/co/cask/gcp/spanner/sink/SpannerSink.java

```diff
@@ -79,10 +114,79 @@ public void prepareRun(BatchSinkContext context) {
 
     List<Schema.Field> fields = config.getSchema().getFields();
     if (fields != null && !fields.isEmpty()) {
-        // Record the field level WriteOperation
-        lineageRecorder.recordWrite("Write", "Wrote to Spanner table.",
-                                    fields.stream().map(Schema.Field::getName).collect(Collectors.toList()));
+      // Record the field level WriteOperation
+      lineageRecorder.recordWrite("Write", "Wrote to Spanner table.",
+                                  fields.stream().map(Schema.Field::getName).collect(Collectors.toList()));
+    }
+  }
+
+  private void createTableIfNotPresent(DatabaseClient dbClient, Database database) {
+    boolean tableExists = isTablePresent(dbClient);
+    if (!tableExists) {
+      String createStmt = SpannerUtil.convertSchemaToCreateStatement(config.table, config.getSchema());
+      LOG.debug("Creating table with create statement: {} in database {} of instance {}", createStmt,
+                config.database, config.instance);
+      // In Spanner table creation is an update ddl operation on the database.
+      Operation<Void, UpdateDatabaseDdlMetadata> op = database.updateDdl(Collections.singletonList(createStmt), null);
+      // Table creation is an async operation. So wait until table is created.
+      op.waitFor().getResult();
+    }
+  }
+
+  private boolean isTablePresent(DatabaseClient dbClient) {
+    // Spanner does not have apis to get table or check if a given table exists. So select the table name from
+    // information schema (metadata) of spanner database.
+    ResultSet resultSet = dbClient
+      .singleUse()
+      .executeQuery(Statement.of("SELECT\n" +
+                                   "    t.table_name\n" +
+                                   "FROM\n" +
+                                   "    information_schema.tables AS t\n" +
+                                   "WHERE\n" +
+                                   "    t.table_catalog = '' AND t.table_schema = '' AND\n" +
+                                   "    t.table_name = '" + config.table + "'"));
```

Comment:
there should be some way to do variable substitution with the spanner library rather than concatenating strings. This is subject to sql injection.

Reply:
Good point